primitives/src/transaction.rs

```diff
@@ -15,6 +15,7 @@ use core::fmt;
 #[cfg(feature = "arbitrary")]
 use arbitrary::{Arbitrary, Unstructured};
 use hashes::sha256d;
+#[cfg(feature = "alloc")]
```

Comment:
Moving `use core::str;` to the test module is good and removes the unused import warning.  But adding this `alloc` feature gate here is unrelated, and from a quick look this is not the only case where it could/should be added.  

Unless there is a specific reason you need it here it might be better to leave it out and address the whole `alloc` feature usage seperately.